src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -90,15 +71,18 @@ internal void BuildWorkerProviderDictionary()
 
         internal void AddProviders()
         {
-            _logger.LogDebug("Workers Directory set to: {WorkersDirPath}", WorkersDirPath);
+            _workerConfigurationResolutionInfo = _workerConfigurationResolver.GetWorkerConfigPaths();
+            var workerConfigs = _workerConfigurationResolutionInfo.WorkerConfigPaths;
 
-            foreach (var workerDir in Directory.EnumerateDirectories(WorkersDirPath))
+            if (workerConfigs is null || !workerConfigs.Any())
             {
-                string workerConfigPath = Path.Combine(workerDir, RpcWorkerConstants.WorkerConfigFileName);
-                if (File.Exists(workerConfigPath))
-                {
-                    AddProvider(workerDir);
-                }
+                _logger.LogTrace("No worker configs found.");
```

Comment:
What happens upstream if no worker configs are found? We just return here so I am wondering if we should throw, is this a bad state?